src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -1461,6 +1543,34 @@ skip:
     });
   }
   //------------------------------------------------------------------------------------------------------------------------
+  // Tells the other end to send us the given txes (typically with attached blink data) as if they
+  // are new transactions.
+  template<class t_core>
+  int t_cryptonote_protocol_handler<t_core>::handle_request_get_txs(int command, NOTIFY_REQUEST_GET_TXS::request& arg, cryptonote_connection_context& context)
+  {
+    MLOG_P2P_MESSAGE("Received NOTIFY_REQUEST_GET_TXS (" << arg.txs.size() << " txs)");
+
+    if (arg.txs.size() > CURRENCY_PROTOCOL_MAX_TXS_REQUEST_COUNT)
+    {
+      LOG_ERROR_CCONTEXT(
+          "Requested txs count is too big (" << arg.txs.size() << ") expected not more than " << CURRENCY_PROTOCOL_MAX_TXS_REQUEST_COUNT);
+      drop_connection(context, false, false);
+      return 1;
+    }
+
+    NOTIFY_NEW_TRANSACTIONS::request rsp;
+    rsp.requested = true;
+    if(!m_core.get_blockchain_storage().handle_get_txs(arg, rsp))
+    {
+      LOG_ERROR_CCONTEXT("failed to handle request NOTIFY_REQUEST_GET_TXS, dropping connection");
+      drop_connection(context, false, false);
+      return 1;
+    }
+    MLOG_P2P_MESSAGE("-->>NOTIFY_NEW_TRANSACTIONS: requested=true, txs[" << rsp.txs.size() << "], blinks[" << rsp.blinks.size() << "]");
+    post_notify<NOTIFY_NEW_TRANSACTIONS>(rsp, context);
```

Comment:
Are we using NOTIFY_NEW_TRANSACTIONS, because possibly at this point- we know our hashes mismatch, so, either, we're missing some of the transactions they have or, they're missing some of the transactions we have so we need to send it over to fill in their gap if they need it

Reply:
Just a misunderstanding.

Reply:
This is coming in the second step, after they've already sent us the transactions they were missing and we're replying with the defaults.  The ping-pong goes like this:

1. <- SYNC from peer info includes blink hashes
- we detect mismatch for heights H1, H2, so:
2. -> NOTIFY_REQUEST_BLOCK_BLINKS to request a list of blink txs for heights [H1, H2]
3. <- NOTIFY_RESPONSE_BLOCK_BLINKS returns us all the peer's blink tx hashes in H1, H2
4. -> NOTIFY_REQUEST_GET_TXS to request any txs from 3. that we are missing
5. <- NOTIFY_NEW_TRANSACTIONS delivers the transactions and blink hashes requested in 4 (along with a `requested` flag so that we know not to relay them).

Reply:
If we were missing some of theirs this will bring us into sync; if they were also missing some of ours then they will notice on the next core sync status we send to them and initiate the ping pong the other way.

Reply:
Thats the every 60s sync you pointed out in the todo if not mistaken.

Reply:
> Thats the every 60s sync you pointed out in the todo if not mistaken.

The todo item was to make the 60s a per-connection timer rather than a single timer that syncs every node all at the same time every 60s.  (I would not be surprised if this big blast of connections with every peer every 60s is partially responsible for things getting bad with more than a few dozen connections).

The sync also occurs on initial connections with a new peer (and so also when starting up).  Aside from that, this is mainly a sort of safety that isn't expected to fire with an established peer: a normal node is going to receive nodes when they initially get broadcast and will almost always agree on the states.